Objects/unicodeobject.c

```diff
@@ -10057,14 +10057,14 @@ do_capitalize(int kind, const void *data, Py_ssize_t length, Py_UCS4 *res, Py_UC
     Py_UCS4 c, mapped[3];
 
     c = PyUnicode_READ(kind, data, 0);
-    n_res = _PyUnicode_ToTitleFull(c, mapped);
+    n_res = PyUnicode_ToTitle(c, mapped, 3);
```

Comment:
You may use `Py_ARRAY_LENGTH(mapped)` instead of copying hardcoded value (`3`).

You may ensure that the function didn't fail, maybe using an assertion: `assert(n_res >= 1);`.